python/lsst/daf/butler/formatters/yamlFormatter.py

```diff
@@ -54,13 +54,48 @@ def _readFile(self, path, pytype=None):
         The `~yaml.UnsafeLoader` is used when parsing the YAML file.
         """
         try:
-            with open(path, "r") as fd:
-                data = yaml.load(fd, Loader=yaml.UnsafeLoader)
+            with open(path, "rb") as fd:
+                data = self._fromBytes(fd.read())
         except FileNotFoundError:
             data = None
 
         return data
 
+    def _fromBytes(self, inMemoryDataset, pytype=None):
+        """Read the bytes object as a python object.
+
+        Parameters
+        ----------
+        inMemoryDataset : `bytes`
+            Bytes object to unserialize.
+        pytype : `class`, optional
+            Not used by this implementation.
+
+        Returns
+        -------
+        data : `object`
+            Either data as Python object read from the pickled string, or None
+            if the string could not be read.
+        """
+        try:
+            data = yaml.load(inMemoryDataset, Loader=yaml.UnsafeLoader)
```

Comment:
Maybe `inMemoryDataset.decode()` here to make it clear we are passing in strings not bytes to the yaml loader.

Reply:
if this is `_fromBytes` method then should we not expect bytes going into the `yaml.load`?

The following seems to work for me just fine so I left it as is:

```
>>> YamlFormatter()._readFile('test.yaml')
{'martin': {'name': "Martin D'vloper", 'job': 'Developer', 'skill': 'Elite'}}
>>> inmem = YamlFormatter()._readFile('test.yaml')
>>> YamlFormatter()._toBytes(inmem)
b"martin:\n  job: Developer\n  name: Martin D'vloper\n  skill: Elite\n"
>>> inbytes = YamlFormatter()._toBytes(inmem)
>>> inbytes = YamlFormatter()._fromBytes(inbytes)
>>> inbytes = YamlFormatter()._toBytes(inmem)
>>> YamlFormatter()._fromBytes(inbytes)
{'martin': {'job': 'Developer', 'name': "Martin D'vloper", 'skill': 'Elite'}}
```